task/transcode.go

```diff
@@ -115,6 +116,24 @@ func fileWriter(size int64) WriteSeekCloser {
 	}
 }
 
+type SegmentCounter struct {
```

Comment:
**nit:** Perhaps name this `SegmentLengthCounter` (or size) otherwise it seems like it's just counting the number of segments seen, not their size.